pipeline-steps/templates/steps/vault-upload.yaml

```diff
@@ -1,26 +1,26 @@
 parameters:
   azureSubscriptionEndpoint: $(azure.subscription.endpoint)
   azureVault: ''
-  keyMapping: []
+  secretName: ''
+  secretValue: ''
 
 steps:
-  - ${{ each pair in parameters.keyMapping }}:
-      - task: AzureCLI@1
-        displayName: "Set vault secrets ${{ pair.secret_name }}"
-        inputs:
-          azureSubscription: ${{ parameters.azureSubscriptionEndpoint }}
-          scriptLocation: "inlineScript"
-          inlineScript: |
-            seachedSecret=$(az keyvault secret list --vault-name ${{ parameters.azureVault }}  --query "[?name=='${{ pair.secret_name }}'].name" -o tsv)
-            varValue='$(${{ pair.env_var_name }})'
-            if [ -z "$seachedSecret" ]; then
-                 echo  "This is a new secret"
-                 az keyvault secret set --name ${{ pair.secret_name }} --vault-name ${{ parameters.azureVault }}  --value '$(${{ pair.env_var_name }})'
+  - task: AzureCLI@1
+    displayName: "Set vault secrets ${{ parameters.secretName }}"
+    inputs:
+      azureSubscription: ${{ parameters.azureSubscriptionEndpoint }}
+      scriptLocation: "inlineScript"
+      inlineScript: |
+        seachedSecret=$(az keyvault secret list --vault-name ${{ parameters.azureVault }}  --query "[?name=='${{ parameters.secretName }}'].name" -o tsv)
+        varValue='${{ parameters.secretValue }}'
```

Comment:
Very minor, but if you think you can do it quickly, try removing the usage of varValue and just directly using parameters.secretValue where varValue is referenced instead, or vice-versa.
If it doesn't work, then can leave as is.